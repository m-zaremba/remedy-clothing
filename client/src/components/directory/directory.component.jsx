import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { createStructuredSelector } from 'reselect';
import { selectDirectorySections } from '../../redux/directory/directory.selectors';
import MenuItem from '../menu-item/menu-item.component';
import { DirectoryMenuContainer } from './directory.styles';

export const Directory = ({ sections }) => {
  return (
    <DirectoryMenuContainer>
      {sections.map(({ title, imageUrl, id, size, linkUrl }) => (
        <MenuItem key={id} title={title} imageUrl={imageUrl} size={size} linkUrl={linkUrl} />
      ))}
    </DirectoryMenuContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  sections: selectDirectorySections,
});

Directory.propTypes = {
  sections: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default connect(mapStateToProps)(Directory);
