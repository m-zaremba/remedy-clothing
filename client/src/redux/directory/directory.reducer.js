export const INITIAL_STATE = {
  sections: [
    {
      title: 'hats',
      imageUrl: '/images/hats.webp',
      id: 1,
      linkUrl: 'shop/hats',
    },
    {
      title: 'jackets',
      imageUrl: '/images/jackets.webp',
      id: 2,
      linkUrl: 'shop/jackets',
    },
    {
      title: 'sneakers',
      imageUrl: '/images/sneakers.webp',
      id: 3,
      linkUrl: 'shop/sneakers',
    },
    {
      title: 'womens',
      imageUrl: '/images/womens.webp',
      size: 'large',
      id: 4,
      linkUrl: 'shop/womens',
    },
    {
      title: 'mens',
      imageUrl: '/images/men.webp',
      size: 'large',
      id: 5,
      linkUrl: 'shop/mens',
    },
  ],
};

const directoryReducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    default:
      return state;
  }
};

export default directoryReducer;
