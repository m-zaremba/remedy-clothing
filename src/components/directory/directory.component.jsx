import React, { Component } from 'react';
import MenuItem from '../menu-item/menu-item.component';
import './directory.styles.scss';

class Directory extends Component {
  constructor() {
    super();
    this.state = {
      sections: [
        {
          title: 'hats',
          imageUrl: '/images/hats.png',
          id: 1,
          linkUrl: 'hats',
        },
        {
          title: 'jackets',
          imageUrl: '/images/jackets.png',
          id: 2,
          linkUrl: '',
        },
        {
          title: 'sneakers',
          imageUrl: '/images/sneakers.png',
          id: 3,
          linkUrl: '',
        },
        {
          title: 'womens',
          imageUrl: '/images/womens.png',
          size: 'large',
          id: 4,
          linkUrl: '',
        },
        {
          title: 'mens',
          imageUrl: '/images/men.png',
          size: 'large',
          id: 5,
          linkUrl: '',
        },
      ],
    };
  }

  render() {
    const { sections } = this.state;
    return (
      <div className="directory-menu">
        {sections.map(({ title, imageUrl, id, size, linkUrl }) => (
          <MenuItem key={id} title={title} imageUrl={imageUrl} size={size} linkUrl={linkUrl} />
        ))}
      </div>
    );
  }
}

export default Directory;
