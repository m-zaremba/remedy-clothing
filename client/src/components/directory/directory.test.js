import React from 'react';
import { shallow } from 'enzyme';
import { Directory } from './directory.component';

describe('testing Directory component', () => {
  it('should render Directory component', () => {
    expect(shallow(<Directory sections={[]} />)).toMatchSnapshot();
  });
});
