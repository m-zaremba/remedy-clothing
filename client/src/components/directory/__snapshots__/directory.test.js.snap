// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Directory component should render Directory component 1`] = `ShallowWrapper {}`;
